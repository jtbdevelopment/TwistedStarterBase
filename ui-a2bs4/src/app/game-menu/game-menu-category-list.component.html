<div *ngIf="games.length > 0" class="category-wrapper">
    <button class="list-group-item btn-group btn btn-group-{{style}}" (click)="toggleCollapse()">
        <span class="fa fa-{{icon}} float-left"></span>
        <span class="float-left">{{category}}</span>
        <span class="badge badge-secondary badge-pill align-middle float-right ml-auto">{{games.length}}</span>
    </button>
    <div [ngbCollapse]="isCollapsed" class="games-collapse">
        <div *ngFor="let game of games;" class="btn-group-vertical btn-group-games">
            <game-menu-game-item [game]="game" [style]="style" class="game-wrapper"></game-menu-game-item>
        </div>
    </div>
</div>
