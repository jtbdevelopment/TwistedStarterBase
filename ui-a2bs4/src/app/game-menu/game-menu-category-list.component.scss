.category-wrapper {
  width: 100%;

  .fa {
    margin-right: 5px;
    margin-top: 4px;
  }

  .list-group-item {
    width: 100%;
    margin-top: 5px;
    -ms-word-wrap: break-word;
    word-wrap: break-word;
    white-space: normal;
    font-weight: bold;
    padding: 10px 10px;
    border: none;
    color: gray;
    .badge {
      margin-top: 4px;
    }
  }

  .games-collapse {
    width: 100%;
    .btn-group-games {
      width: 100%;
      .game-wrapper {
        width: 100%;
      }
    }
  }
}

