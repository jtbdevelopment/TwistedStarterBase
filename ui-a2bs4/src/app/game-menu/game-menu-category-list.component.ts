import {Component, Input, OnInit} from '@angular/core';
import {Game, GameCacheService} from 'jtb-core-games-ui';


@Component({
  selector: 'app-game-menu-category-list',
  templateUrl: './game-menu-category-list.component.html',
  styleUrls: ['./game-menu-category-list.component.scss']
})
export class GameMenuCategoryListComponent implements OnInit {
  @Input() public category: string;
  @Input() public icon: string;
  @Input() public style: string;
  public games: Game[] = [];
  public isCollapsed: boolean = false;

  constructor(private gameCache: GameCacheService) {
  }

  ngOnInit() {
    this.gameCache.getGamesForCategory(this.category).subscribe(g => {
      this.games = Object.assign([], g);
      this.games.sort((a, b) => {
        if (a.lastUpdate > b.lastUpdate) {
          return 1;
        }
        if (a.lastUpdate === b.lastUpdate) {
          return 0;
        }
        return -1;
      });
    });
  }

  public toggleCollapse(): void {
    this.isCollapsed = !this.isCollapsed;
  }
}
