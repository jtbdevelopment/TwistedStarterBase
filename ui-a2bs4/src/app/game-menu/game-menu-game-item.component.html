<a [routerLink]="[game.standardLink()]"
   class="btn-group btn btn-light btn-group-game btn-group-game-{{style}}">
  <img *ngFor="let md5 of game.players | mapKeys" class="player-image" src="{{game.playerImages[md5]}}">
  <span class="game-description">{{describeGame()}}</span>
</a>

