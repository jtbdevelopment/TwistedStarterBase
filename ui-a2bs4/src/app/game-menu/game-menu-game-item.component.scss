@import '~bootstrap/scss/bootstrap.scss';

.btn-group-game {
  width: 100%;
  padding: 4px;
  alignment: left;
  text-align: left;
  -ms-word-wrap: break-word;
  word-wrap: break-word;
  white-space: normal;
  font-weight: bold;
  border: 1px solid;
  .game-description {
    font-weight: bold;
  }

  .player-image {
    margin: 3px 1px;
    width: 25px;
    height: 25px;
    @include media-breakpoint-down(sm) {
      width: 20px;
      height: 20px;
    }
  }
}
