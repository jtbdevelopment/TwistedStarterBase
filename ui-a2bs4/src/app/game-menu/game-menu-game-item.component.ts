import {Component, Input} from '@angular/core';
import {Game} from '../core-ui/games/game.model';
@Component({
    selector: 'game-menu-game-item',
    template: require('./game-menu-game-item.component.html')
})
export class GameMenuGameItemComponent {
    @Input() public game: Game;
    @Input() public style: string;

    //  TODO - TSB
    public describeGame(): string {
        return 'TODO';
    }
}
