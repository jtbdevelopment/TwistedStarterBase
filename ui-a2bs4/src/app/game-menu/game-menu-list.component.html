<div class="game-bar">
    <div *ngFor="let category of categories; let i = index;"
         class="list-group list-group-games list-group-{{styles[i]}}">
        <game-menu-category-list [category]="category"
                                 [icon]="icons.get(category)"
                                 [style]="styles[i]">
        </game-menu-category-list>
    </div>
</div>