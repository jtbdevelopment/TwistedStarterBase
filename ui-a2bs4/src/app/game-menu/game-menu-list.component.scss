@import '~bootstrap/scss/bootstrap.scss';

.game-bar {
  padding-left: 5px;

  .list-group-games {
    margin: 0;
    .category-wrapper {
      width: 100%;

      .fa {
        margin-right: 5px;
      }

      .list-group-item {
        width: 100%;
        margin-top: 5px;
        -ms-word-wrap: break-word;
        word-wrap: break-word;
        white-space: normal;
        font-weight: bold;
        padding: 10px 10px;
        border: none;
      }

      .games-collapse {
        width: 100%;
        .btn-group-games {
          width: 100%;

          .game-wrapper {
            width: 100%;

            .btn-group-game {
              width: 100%;
              padding: 4px;
              alignment: left;
              text-align: left;
              -ms-word-wrap: break-word;
              word-wrap: break-word;
              white-space: normal;
              font-weight: bold;
              .game-description {
                font-weight: bold;
              }

              .player-image {
                margin: 3px 1px;
                width: 25px;
                height: 25px;
                @include media-breakpoint-down(sm) {
                  width: 20px;
                  height: 20px;
                }
              }
            }
          }
        }
      }
    }
  }
}