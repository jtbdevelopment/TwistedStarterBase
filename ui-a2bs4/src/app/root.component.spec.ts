import {Component} from '@angular/core';
import {inject, TestBed} from '@angular/core/testing';
import {RootComponent} from './root.component';
import {RouterTestingModule} from '@angular/router/testing';
import {BehaviorSubject} from 'rxjs/BehaviorSubject';
import {GameMenuService} from './game-menu/game-menu.service';

@Component({selector: 'app-navigation-bar', template: '<div class="nb"></div>'})
class MockNavBarComponent {
}

@Component({selector: 'app-game-menu-list', template: '<div class="gml"></div>'})
class MockGameMenuComponent {
}

@Component(
  {
    selector: 'router-outlet',     // tslint:disable-line
    template: '<div class="ro"></div>'
  }
)
class MockRouterOutletComponent {
}

class MockGameMenuService {
  public showGames: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
}


describe('Root Component', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [RouterTestingModule],
      providers: [
        {provide: GameMenuService, useClass: MockGameMenuService}
      ],
      declarations: [
        RootComponent,
        MockNavBarComponent,
        MockGameMenuComponent,
        MockRouterOutletComponent
      ]
    });
    TestBed.compileComponents();
  });

  it('should render the shell when no show games', () => {
    const fixture = TestBed.createComponent(RootComponent);
    fixture.detectChanges();
    const main = fixture.nativeElement;
    expect(main.querySelector('.nb')).not.toBeNull();
    expect(main.querySelector('.gml')).toBeNull();
    expect(main.querySelector('.ro')).not.toBeNull();
    expect(main.querySelector('.col-12')).not.toBeNull();
    expect(main.querySelector('.col-10')).toBeNull();
  });

  it('should render the shell when show games', inject([GameMenuService], (gameMenuService) => {
    const fixture = TestBed.createComponent(RootComponent);
    gameMenuService.showGames.next(true);
    fixture.detectChanges();

    const main = fixture.nativeElement;
    expect(main.querySelector('.nb')).not.toBeNull();
    expect(main.querySelector('.gml')).not.toBeNull();
    expect(main.querySelector('.ro')).not.toBeNull();
    expect(main.querySelector('.col-12')).toBeNull();
    expect(main.querySelector('.col-10')).not.toBeNull();
  }));
});
