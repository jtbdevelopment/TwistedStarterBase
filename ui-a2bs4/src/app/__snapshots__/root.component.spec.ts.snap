// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Component should render the shell when no show games 1`] = `
<app-root
  showGames="false"
>
  <div
    class="container-fluid"
  >
    
    
    <div
      class="row"
    >
      
      
      
      
      
      <div
        class="col-12"
        ng-reflect-ng-class="[object Object]"
      >
        
        
        <router-outlet>
          <div
            class="ro"
          />
        </router-outlet>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  <app-navigation-bar>
    <div
      class="nb"
    />
  </app-navigation-bar>
  
</app-root>
`;

exports[`Root Component should render the shell when show games 1`] = `
<app-root
  showGames={[Function Boolean]}
>
  <div
    class="container-fluid"
  >
    
    
    <div
      class="row"
    >
      
      
      
      <div
        class="col-2"
      >
        
        
        <app-game-menu-list>
          <div
            class="gml"
          />
        </app-game-menu-list>
        
      
      </div>
      
      
      <div
        class="col-10"
        ng-reflect-ng-class="[object Object]"
      >
        
        
        <router-outlet>
          <div
            class="ro"
          />
        </router-outlet>
        
      
      </div>
      
    
    </div>
    
  
  </div>
  <app-navigation-bar>
    <div
      class="nb"
    />
  </app-navigation-bar>
  
</app-root>
`;
